### ISSUE TITLE
Build failing without error message

### ISSUE THREAD

**Issue by rsteckler** [Shippable/support#2205]
My builds are failing when I try to deploy to beanstalk.  I'm sure it's an error somewhere on my configuration, but the build console doesn't show any errors.  The overall build fails, but each step shows a green checkmark without any indication of what's wrong.  I tried downloading the build log, but it didn't have any additional information.  

When I remove the Deployment config in the shippable UI, the build succeeds, so I'm guessing the problem is related to that piece.  **Is there any way to figure out what's wrong with my config?**

Here's a failed build, if that helps for debugging:  56970ba259dc660c0050dff4

Thanks,
Ryan

**Comment by rsteckler** [comment]
Some more details.

Here is my top-level log, so you can see which tasks are happening:
```
  Generating builds 0.00 seconds
  hub login 0.54 seconds
  pulling image from hub 5.56 seconds
  setup_jdk 0.22 seconds
  setup_ve 3.58 seconds
  start_services 14.44 seconds
  git_sync 10.00 seconds
  git_metadata 0.16 seconds
  before_install 6.05 seconds
  install 163.61 seconds
  script 20.07 seconds
  save reports 0.00 seconds
  commit container 13.89 seconds
  hub login 0.63 seconds
  hub push 273.40 seconds
  purging build container 4.02 seconds
```

Here is my Dockerrun.aws.json file:
```
  {
   "AWSEBDockerrunVersion": "1",
   "Image": {
     "Name": "<IMAGE_NAME>:<TAG>",
     "Update": "true"
   },
   "Ports": [
     {
       "ContainerPort": "3000"
     }
   ],
  "Authentication": {
     "Bucket": "elasticbeanstalk-us-east-1-fancy_long_number",
     "Key": "<DOCKER_CONFIG>"
   }
  }
```

My shippable.yml
```
language: node_js

# no node version specifier because it causes shippable to build node from source which takes FOREVER
# node_js:
#   - 0.12.7

before_install:
  - mkdir -p ~/.nvm/versions
  - source ~/.nvm/nvm.sh && nvm install 0.12.7

services:
  - mongodb

after_success:
  # - ./restartifneeded.sh
  
branches:
  only:
    - master
```

**Comment by rsteckler** [comment]
Ok.  I'm much further along now.  There's a bunch of things I had to change in the Dockerfile, Dockerrun.aws.json, and the project settings.  

I'm getting a new error now in the build console:
```
A client error (SignatureDoesNotMatch) occurred when calling the PutObject operation: The request signature we calculated does not match the signature you provided. Check your key and signing method.
```

...but at least I'm getting an error so I can go take a look.  For the Shippable team, I'd recommend two things:
1)  The documentation doesn't match the UI anymore with regards to Docker builds.  There aren't options for pre- or post- CI, etc.  
2)  An updated how-to with a sample project would go a long way.  Things like "source directory" were confusing;  I thought that's where shippable was going to copy my code from git, but it's really where it expects it to be, etc.

**Comment by rsteckler** [comment]
K.  The SignatureDoesNotMatch error was due to something in the AWS key I setup with Shippable.  Simply creating a new key worked.  According to some issues at the aws-cli, + signs can be urlencoded incorrectly in requests, causing this error.  I just created some new keys until I got one that didn't have any pluses, slashes, etc.  Now the build is succeeding. 

...but the AWS EBS event log has an error.  Down the rabbit hole we go.
